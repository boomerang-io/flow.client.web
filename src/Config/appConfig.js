// Look for the data injected into the HTML file from the Express app
// See server/app.js for implementation
export const APP_ROOT =
  window._SERVER_DATA && window._SERVER_DATA.APP_ROOT ? window._SERVER_DATA.APP_ROOT : "/BMRG_APP_ROOT_CONTEXT";

export const PRODUCT_STANDALONE =
  window._SERVER_DATA && window._SERVER_DATA.PRODUCT_STANDALONE
    ? window._SERVER_DATA.PRODUCT_STANDALONE === "true"
    : process.env.REACT_APP_PRODUCT_STANDALONE
    ? true
    : false;

export const EMBEDDED_MODE =
  window._SERVER_DATA && window._SERVER_DATA.EMBEDDED_MODE
    ? window._SERVER_DATA.EMBEDDED_MODE === "true"
    : process.env.REACT_APP_EMBEDDED_MODE
    ? true
    : true;

export const CORE_ENV_URL =
  window._SERVER_DATA && window._SERVER_DATA.CORE_ENV_URL ? window._SERVER_DATA.CORE_ENV_URL : "";

export const BASE_DOCUMENTATION_URL = PRODUCT_STANDALONE
  ? "https://www.useboomerang.io/docs/boomerang-flow"
  : `${CORE_ENV_URL}/docs/boomerang-flow`;

export const isDevEnv = process.env.NODE_ENV === "development";
export const isTestEnv = process.env.NODE_ENV === "test";

export const AppPath = {
  Root: "/",
  Error: "/error",
  Activity: "/activity",
  Editor: "/editor/:workflowId",
  EditorDesigner: `/editor/:workflowId/workflow`,
  EditorConfigure: `editor/:workflowId/configure`,
  EditorChangelog: `/editor/:workflowId/changelog`,
  EditorProperties: `/editor/:workflowId/parameters`,
  Execution: "/activity/:workflowId/execution/:executionId",
  Insights: "/insights",
  Properties: "/admin/parameters",
  Quotas: "/admin/quotas",
  QuotasEdit: "/admin/quotas/:teamId",
  Settings: "/admin/settings",
  SystemWorkflows: "/admin/system-workflows",
  TaskTemplates: "/admin/task-templates",
  TaskTemplateEdit: `/admin/task-templates/:id/:version`,
  TaskTemplateYaml: `/admin/task-templates/:id/:version/yaml-editor`,
  Team: "/admin/teams/:teamId",
  TeamSettings: "/admin/teams/:teamId/settings",
  TeamWorkflows: "/admin/teams/:teamId/workflows",
  TeamList: "/admin/teams",
  TeamProperties: `/manage/team-parameters`,
  User: "/admin/users/:userId",
  UserList: "/admin/users",

  Workflows: "/workflows",
  WorkflowsMine: "/workflows/mine",
  WorkflowsTeams: "/workflows/teams",

  ManageTaskTemplates: `/manage/task-templates`,
  ManageTaskTemplatesTeam: `/manage/task-templates/team/:teamId`,
  ManageTaskTemplateEdit: `/manage/task-templates/team/:teamId/:taskId/:version`,
  ManageTaskTemplateYaml: `/manage/task-templates/team/:teamId/:taskId/:version/yaml-editor`,
};

export const appLink = {
  activity: () => `/activity`,
  editorDesigner: ({ workflowId }) => `/editor/${workflowId}/workflow`,
  editorConfigure: ({ workflowId }) => `/editor/${workflowId}/configure`,
  editorChangelog: ({ workflowId }) => `/editor/${workflowId}/changelog`,
  editorProperties: ({ workflowId }) => `/editor/${workflowId}/parameters`,
  execution: ({ executionId, workflowId }) => `/activity/${workflowId}/execution/${executionId}`,
  insights: () => "/insights",
  properties: () => "/admin/parameters",
  manageUsers: () => "/admin/users",
  quotas: () => "/admin/quotas",
  quotasEdit: ({ teamId }) => `/admin/quotas/${teamId}`,
  settings: () => "/admin/settings",
  systemWorkflows: () => "/admin/system-workflows",
  taskTemplates: () => "/admin/task-templates",
  taskTemplateEdit: ({ id, version }) => `/admin/task-templates/${id}/${version}`,
  taskTemplateYaml: ({ id, version }) => `/admin/task-templates/${id}/${version}/yaml-editor`,
  teamProperties: () => `/manage/team-parameters`,
  teamTaskTemplates: () => `/manage/task-templates`,
  team: ({ teamId }) => `/admin/teams/${teamId}`,
  teamWorkflows: ({ teamId }) => `/admin/teams/${teamId}/workflows`,
  teamSettings: ({ teamId }) => `/admin/teams/${teamId}/settings`,
  teamList: () => "/admin/teams",
  workflows: () => "/workflows",
  workflowsMine: () => "/workflows/mine",
  workflowsTeams: () => "/workflows/teams",
  workflowActivity: ({ workflowId }) => `/activity?page=0&size=10&workflowIds=${workflowId}`,

  manageTaskTemplates: ({ teamId }) => `/manage/task-templates/team/${teamId}`,
  manageTaskTemplateEdit: ({ teamId, taskId, version }) => `/manage/task-templates/team/${teamId}/${taskId}/${version}`,
  manageTaskTemplateYaml: ({ teamId, taskId, version }) =>
    `/manage/task-templates/team/${teamId}/${taskId}/${version}/yaml-editor`,

  //external apps
  docsWorkflowEditor: () => `${BASE_DOCUMENTATION_URL}/how-to-guide/workflow-editor`,
};

export const queryStringOptions = { arrayFormat: "comma", skipEmptyString: true };

export const FeatureFlag = {
  /**
   * new Feature Flags
   */
  TeamManagementEnabled: "TeamManagementEnabled",
  WorkflowQuotasEnabled: "WorkflowQuotasEnabled",
  SettingsEnabled: "SettingsEnabled",
  UserManagementEnabled: "UserManagementEnabled",
  GlobalParametersEnabled: "GlobalParametersEnabled",
  WorkflowTokensEnabled: "WorkflowTokensEnabled",
  TaskManagerEnabled: "TaskManagerEnabled",
  EditVerifiedTasksEnabled: "EditVerifiedTasksEnabled",
  WorkflowTriggersEnabled: "WorkflowTriggersEnabled",
  TeamParametersEnabled: "TeamParametersEnabled",

  ActivityEnabled: "ActivityEnabled",
  InsightsEnabled: "InsightsEnabled",
};
