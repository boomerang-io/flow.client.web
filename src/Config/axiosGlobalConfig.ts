// Setup global config for Axios requests so they work with ICP
import axios from "axios";
axios.defaults.withCredentials = true;
