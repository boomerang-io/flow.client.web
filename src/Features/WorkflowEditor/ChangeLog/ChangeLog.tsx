import React from "react";
import { Helmet } from "react-helmet";
import ChangeLogTable from "./ChangeLogTable";
import { ChangeLog as ChangeLogType } from "Types";
import styles from "./changeLog.module.scss";

interface ChangeLogProps {
  changeLogData: ChangeLogType;
}

function ChangeLog({ changeLogData }: ChangeLogProps) {
  return (
    <div className={styles.container}>
      <Helmet>
        <title>Change Log</title>
      </Helmet>
      <ChangeLogTable changeLog={changeLogData} />
    </div>
  );
}

export default ChangeLog;
