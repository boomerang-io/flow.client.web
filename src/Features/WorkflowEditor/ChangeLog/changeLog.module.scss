.container {
  display: flex;
  flex-direction: column;
  margin: 0 auto;
  overflow: auto;
  padding: 2rem 2rem 1rem;
  width: 100%;

  :global .bmrg--b-loading-animation__text {
    display: none;
  }
  :global .bmrg--b-loading-animation__img {
    transform: scale(0.3);
  }
}

.searchSkeleton {
  margin-bottom: 1rem;
  width: 300px;
}
