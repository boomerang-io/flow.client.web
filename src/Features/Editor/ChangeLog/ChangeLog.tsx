import React from "react";
import { Helmet } from "react-helmet";
import { useQuery } from "Hooks";
import { DelayedRender, DataTableSkeleton, SearchSkeleton } from "@boomerang-io/carbon-addons-boomerang-react";
import ErrorDragon from "Components/ErrorDragon";
import ChangeLogTable from "./ChangeLogTable";
import qs from "query-string";
import { serviceUrl } from "Config/servicesConfig";
import { WorkflowSummary } from "Types";
import styles from "./changeLog.module.scss";

interface ChangeLogProps {
  summaryData: WorkflowSummary;
}

const ChangeLog: React.FC<ChangeLogProps> = ({ summaryData }) => {
  const getWorkflowChangelogUrl = serviceUrl.getWorkflowChangelog({
    workflowId: summaryData.id,
    query: qs.stringify({ sort: "version", order: "DESC" }),
  });
  const { data, error, isLoading } = useQuery(getWorkflowChangelogUrl);
  if (isLoading)
    return (
      <DelayedRender>
        <div className={styles.container}>
          <div className={styles.searchSkeleton}>
            <SearchSkeleton small />
          </div>
          <DataTableSkeleton />
        </div>
      </DelayedRender>
    );

  if (error) return <ErrorDragon />;

  return (
    <div className={styles.container}>
      <Helmet>
        <title>{`Change Log - ${summaryData.name}`}</title>
      </Helmet>
      <ChangeLogTable changeLog={data} />
    </div>
  );
};

export default ChangeLog;
