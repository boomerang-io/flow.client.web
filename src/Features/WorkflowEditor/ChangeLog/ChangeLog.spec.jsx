import React from "react";
import ChangeLog from ".";
import { Route } from "react-router-dom";
import { screen } from "@testing-library/react";
import { AppPath, appLink } from "Config/appConfig";
import { startApiServer } from "ApiServer";

let server;

beforeEach(() => {
  server = startApiServer();
});

afterEach(() => {
  server.shutdown();
});

const props = {
  summaryData: {
    id: "5eb2c4085a92d80001a16d87",
    name: "Test name",
  },
};

describe("ChangeLog --- Snapshot Test", () => {
  it("Capturing Snapshot of ChangeLog", async () => {
    const { baseElement } = rtlContextRouterRender(
      <Route path={AppPath.EditorChangelog}>
        <ChangeLog {...props} />
      </Route>,
      { route: appLink.editorChangelog({ workflowId: "5eb2c4085a92d80001a16d87" }) }
    );
    await screen.findByText("hello sir");
    expect(baseElement).toMatchSnapshot();
  });
});
