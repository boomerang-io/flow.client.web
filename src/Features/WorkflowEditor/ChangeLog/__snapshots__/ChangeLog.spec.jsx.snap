// Vitest Snapshot v1

exports[`ChangeLog --- Snapshot Test > Capturing Snapshot of ChangeLog 1`] = `
<body
  id="app"
>
  <div>
    <div
      class="_container_b75634"
    >
      <div
        class="_tableContainer_468963"
      >
        <div
          aria-labelledby="change-log-table-search-search"
          class="cds--search cds--search--md _search_468963"
          role="search"
        >
          <div
            class="cds--search-magnifier"
          >
            <svg
              aria-hidden="true"
              class="cds--search-magnifier-icon"
              fill="currentColor"
              focusable="false"
              height="16"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15,14.3L10.7,10c1.9-2.3,1.6-5.8-0.7-7.7S4.2,0.7,2.3,3S0.7,8.8,3,10.7c2,1.7,5,1.7,7,0l4.3,4.3L15,14.3z M2,6.5 C2,4,4,2,6.5,2S11,4,11,6.5S9,11,6.5,11S2,9,2,6.5z"
              />
            </svg>
          </div>
          <label
            class="cds--label"
            for="change-log-table-search"
            id="change-log-table-search-search"
          >
            Search
          </label>
          <input
            autocomplete="off"
            class="cds--search-input"
            data-testid="change-log-search"
            id="change-log-table-search"
            placeholder="Search"
            role="searchbox"
            type="text"
            value=""
          />
          <button
            aria-label="Clear search input"
            class="cds--search-close cds--search-close--hidden"
            title="Clear search input"
            type="button"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              height="16"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 32 32"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M24 9.4L22.6 8 16 14.6 9.4 8 8 9.4 14.6 16 8 22.6 9.4 24 16 17.4 22.6 24 24 22.6 17.4 16 24 9.4z"
              />
            </svg>
          </button>
        </div>
        <div
          class="cds--data-table-container"
        >
          <div
            class="cds--data-table-content"
          >
            <table
              class="cds--data-table cds--data-table--sort"
            >
              <thead>
                <tr>
                  <th
                    aria-sort="none"
                    id="version"
                    scope="col"
                  >
                    <div
                      id="table-sort-2"
                      style="display: none;"
                    >
                      Click to sort rows by Version header in ascending order
                    </div>
                    <button
                      aria-describedby="table-sort-2"
                      class="cds--table-sort"
                      type="button"
                    >
                      <span
                        class="cds--table-sort__flex"
                      >
                        <div
                          class="cds--table-header-label"
                        >
                          Version
                        </div>
                        <svg
                          aria-hidden="true"
                          class="cds--table-sort__icon"
                          fill="currentColor"
                          focusable="false"
                          height="20"
                          preserveAspectRatio="xMidYMid meet"
                          viewBox="0 0 32 32"
                          width="20"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M16 4L6 14 7.41 15.41 15 7.83 15 28 17 28 17 7.83 24.59 15.41 26 14 16 4z"
                          />
                        </svg>
                        <svg
                          aria-hidden="true"
                          class="cds--table-sort__icon-unsorted"
                          fill="currentColor"
                          focusable="false"
                          height="20"
                          preserveAspectRatio="xMidYMid meet"
                          viewBox="0 0 32 32"
                          width="20"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M27.6 20.6L24 24.2 24 4 22 4 22 24.2 18.4 20.6 17 22 23 28 29 22zM9 4L3 10 4.4 11.4 8 7.8 8 28 10 28 10 7.8 13.6 11.4 15 10z"
                          />
                        </svg>
                      </span>
                    </button>
                  </th>
                  <th
                    aria-sort="none"
                    id="userName"
                    scope="col"
                  >
                    <div
                      id="table-sort-3"
                      style="display: none;"
                    >
                      Click to sort rows by User header in ascending order
                    </div>
                    <button
                      aria-describedby="table-sort-3"
                      class="cds--table-sort"
                      type="button"
                    >
                      <span
                        class="cds--table-sort__flex"
                      >
                        <div
                          class="cds--table-header-label"
                        >
                          User
                        </div>
                        <svg
                          aria-hidden="true"
                          class="cds--table-sort__icon"
                          fill="currentColor"
                          focusable="false"
                          height="20"
                          preserveAspectRatio="xMidYMid meet"
                          viewBox="0 0 32 32"
                          width="20"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M16 4L6 14 7.41 15.41 15 7.83 15 28 17 28 17 7.83 24.59 15.41 26 14 16 4z"
                          />
                        </svg>
                        <svg
                          aria-hidden="true"
                          class="cds--table-sort__icon-unsorted"
                          fill="currentColor"
                          focusable="false"
                          height="20"
                          preserveAspectRatio="xMidYMid meet"
                          viewBox="0 0 32 32"
                          width="20"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M27.6 20.6L24 24.2 24 4 22 4 22 24.2 18.4 20.6 17 22 23 28 29 22zM9 4L3 10 4.4 11.4 8 7.8 8 28 10 28 10 7.8 13.6 11.4 15 10z"
                          />
                        </svg>
                      </span>
                    </button>
                  </th>
                  <th
                    aria-sort="none"
                    id="reason"
                    scope="col"
                  >
                    <div
                      id="table-sort-4"
                      style="display: none;"
                    >
                      Click to sort rows by Reason header in ascending order
                    </div>
                    <button
                      aria-describedby="table-sort-4"
                      class="cds--table-sort"
                      type="button"
                    >
                      <span
                        class="cds--table-sort__flex"
                      >
                        <div
                          class="cds--table-header-label"
                        >
                          Reason
                        </div>
                        <svg
                          aria-hidden="true"
                          class="cds--table-sort__icon"
                          fill="currentColor"
                          focusable="false"
                          height="20"
                          preserveAspectRatio="xMidYMid meet"
                          viewBox="0 0 32 32"
                          width="20"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M16 4L6 14 7.41 15.41 15 7.83 15 28 17 28 17 7.83 24.59 15.41 26 14 16 4z"
                          />
                        </svg>
                        <svg
                          aria-hidden="true"
                          class="cds--table-sort__icon-unsorted"
                          fill="currentColor"
                          focusable="false"
                          height="20"
                          preserveAspectRatio="xMidYMid meet"
                          viewBox="0 0 32 32"
                          width="20"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M27.6 20.6L24 24.2 24 4 22 4 22 24.2 18.4 20.6 17 22 23 28 29 22zM9 4L3 10 4.4 11.4 8 7.8 8 28 10 28 10 7.8 13.6 11.4 15 10z"
                          />
                        </svg>
                      </span>
                    </button>
                  </th>
                  <th
                    aria-sort="none"
                    id="date"
                    scope="col"
                  >
                    <div
                      id="table-sort-5"
                      style="display: none;"
                    >
                      Click to sort rows by Date header in ascending order
                    </div>
                    <button
                      aria-describedby="table-sort-5"
                      class="cds--table-sort"
                      type="button"
                    >
                      <span
                        class="cds--table-sort__flex"
                      >
                        <div
                          class="cds--table-header-label"
                        >
                          Date
                        </div>
                        <svg
                          aria-hidden="true"
                          class="cds--table-sort__icon"
                          fill="currentColor"
                          focusable="false"
                          height="20"
                          preserveAspectRatio="xMidYMid meet"
                          viewBox="0 0 32 32"
                          width="20"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M16 4L6 14 7.41 15.41 15 7.83 15 28 17 28 17 7.83 24.59 15.41 26 14 16 4z"
                          />
                        </svg>
                        <svg
                          aria-hidden="true"
                          class="cds--table-sort__icon-unsorted"
                          fill="currentColor"
                          focusable="false"
                          height="20"
                          preserveAspectRatio="xMidYMid meet"
                          viewBox="0 0 32 32"
                          width="20"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M27.6 20.6L24 24.2 24 4 22 4 22 24.2 18.4 20.6 17 22 23 28 29 22zM9 4L3 10 4.4 11.4 8 7.8 8 28 10 28 10 7.8 13.6 11.4 15 10z"
                          />
                        </svg>
                      </span>
                    </button>
                  </th>
                </tr>
              </thead>
              <tbody
                aria-live="polite"
                class="_tableBody_468963"
              >
                <tr
                  data-testid="change-log-table-row"
                >
                  <td>
                    <div
                      class="_tableCell_468963"
                    >
                      <p
                        class="_tableText_468963"
                      >
                        2
                      </p>
                    </div>
                  </td>
                  <td>
                    <div
                      class="_tableCell_468963"
                    >
                      <p
                        class="_tableText_468963"
                      >
                        Boomerang/chicago/ibm Boomerang/chicago/ibm
                      </p>
                    </div>
                  </td>
                  <td>
                    <div
                      class="_tableCell_468963"
                    >
                      <p
                        class="_tableText_468963"
                      >
                        Sorry. Removing changes.
                      </p>
                    </div>
                  </td>
                  <td>
                    <div
                      class="_tableCell_468963"
                    >
                      <p
                        class="_tableText_468963"
                      >
                        2020-05-28 07:32 PM
                      </p>
                    </div>
                  </td>
                </tr>
                <tr
                  data-testid="change-log-table-row"
                >
                  <td>
                    <div
                      class="_tableCell_468963"
                    >
                      <p
                        class="_tableText_468963"
                      >
                        1
                      </p>
                    </div>
                  </td>
                  <td>
                    <div
                      class="_tableCell_468963"
                    >
                      <p
                        class="_tableText_468963"
                      >
                        Boomerang/chicago/ibm Boomerang/chicago/ibm
                      </p>
                    </div>
                  </td>
                  <td>
                    <div
                      class="_tableCell_468963"
                    >
                      <p
                        class="_tableText_468963"
                      >
                        hello sir
                      </p>
                    </div>
                  </td>
                  <td>
                    <div
                      class="_tableCell_468963"
                    >
                      <p
                        class="_tableText_468963"
                      >
                        2020-05-28 05:42 PM
                      </p>
                    </div>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <div
          class="cds--pagination cds--pagination--md"
        >
          <div
            class="cds--pagination__left"
          >
            <label
              class="cds--pagination__text"
              for="cds-pagination-select-id-11"
              id="cds-pagination-select-id-11-count-label"
            >
              Items per page:
            </label>
            <div
              class="cds--form-item cds--select__item-count"
            >
              <div
                class="cds--select cds--select--inline"
              >
                <div
                  class="cds--select-input--inline__wrapper"
                >
                  <div
                    class="cds--select-input__wrapper"
                  >
                    <select
                      class="cds--select-input"
                      id="cds-pagination-select-id-11"
                    >
                      <option
                        class="cds--select-option"
                        value="10"
                      >
                        10
                      </option>
                      <option
                        class="cds--select-option"
                        value="25"
                      >
                        25
                      </option>
                      <option
                        class="cds--select-option"
                        value="50"
                      >
                        50
                      </option>
                    </select>
                    <svg
                      aria-hidden="true"
                      class="cds--select__arrow"
                      fill="currentColor"
                      focusable="false"
                      height="16"
                      preserveAspectRatio="xMidYMid meet"
                      viewBox="0 0 16 16"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
            <span
              class="cds--pagination__text cds--pagination__items-count"
            >
              1–2 of 2 items
            </span>
          </div>
          <div
            class="cds--pagination__right"
          >
            <div
              class="cds--form-item cds--select__page-number"
            >
              <div
                class="cds--select cds--select--inline"
              >
                <label
                  class="cds--label cds--visually-hidden"
                  for="cds-pagination-select-id-11-right"
                >
                  Page number, of 1 pages
                </label>
                <div
                  class="cds--select-input--inline__wrapper"
                >
                  <div
                    class="cds--select-input__wrapper"
                  >
                    <select
                      class="cds--select-input"
                      id="cds-pagination-select-id-11-right"
                    >
                      <option
                        class="cds--select-option"
                        value="1"
                      >
                        1
                      </option>
                    </select>
                    <svg
                      aria-hidden="true"
                      class="cds--select__arrow"
                      fill="currentColor"
                      focusable="false"
                      height="16"
                      preserveAspectRatio="xMidYMid meet"
                      viewBox="0 0 16 16"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
            <span
              class="cds--pagination__text"
            >
              of 1 page
            </span>
            <div
              class="cds--pagination__control-buttons"
            >
              <span
                class="cds--popover-container cds--popover--caret cds--popover--high-contrast cds--popover--top cds--tooltip cds--icon-tooltip"
              >
                <button
                  aria-labelledby="tooltip-8"
                  class="cds--btn--icon-only cds--pagination__button cds--pagination__button--backward cds--pagination__button--no-index cds--btn cds--btn--md cds--btn--ghost cds--btn--disabled"
                  disabled=""
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    focusable="false"
                    height="16"
                    preserveAspectRatio="xMidYMid meet"
                    viewBox="0 0 32 32"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M20 24L10 16 20 8z"
                    />
                  </svg>
                </button>
                <span
                  aria-hidden="true"
                  class="cds--popover"
                  id="tooltip-8"
                  role="tooltip"
                >
                  <span
                    class="cds--popover-content cds--tooltip-content"
                  >
                    Previous page
                  </span>
                  <span
                    class="cds--popover-caret"
                  />
                </span>
              </span>
              <span
                class="cds--popover-container cds--popover--caret cds--popover--high-contrast cds--popover--top-right cds--tooltip cds--icon-tooltip"
              >
                <button
                  aria-labelledby="tooltip-10"
                  class="cds--btn--icon-only cds--pagination__button cds--pagination__button--forward cds--pagination__button--no-index cds--btn cds--btn--md cds--btn--ghost cds--btn--disabled"
                  disabled=""
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    focusable="false"
                    height="16"
                    preserveAspectRatio="xMidYMid meet"
                    viewBox="0 0 32 32"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12 8L22 16 12 24z"
                    />
                  </svg>
                </button>
                <span
                  aria-hidden="true"
                  class="cds--popover"
                  id="tooltip-10"
                  role="tooltip"
                >
                  <span
                    class="cds--popover-content cds--tooltip-content"
                  >
                    Next page
                  </span>
                  <span
                    class="cds--popover-caret"
                  />
                </span>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
